File: frontend/src/types/site-content.ts
/** 首页文案（来自 /site-content） */

export interface HeroContent {
  title: string
  description: string
}

/** 快捷入口兜底文案：实际首页入口优先使用 site_settings.home_quick_entries */
export interface QuickEntryContent {
  key: string
  title: string
  desc: string
}

export interface GuideStepContent {
  title: string
  desc: string
}

/** 注册策略提示文案：开关由后台 site-settings 控制，这里仅提供关闭时的提示文案与管理员联系方式 */
export interface RegistrationContent {
  /** 注册关闭时的提示文案 */
  disabled_message: string
  /** 暂停注册时提示联系的管理员（可空） */
  admin_contact: string
}

/** 账户未激活提示文案：登录未激活账户时弹窗展示。
 *  支持点击计数彩蛋：第 1-2 次正常版，第 3 次起着急版，第 5 次触发猫爪彩蛋。 */
export interface ActivationContent {
  /** 未激活弹窗的正文文案（正常版） */
  message: string
  /** 联系管理员的方式（可空，后端缺省时回退到 registration.admin_contact） */
  admin_contact: string
  /** 标题（正常版） */
  title: string
  /** 按钮文案（正常版） */
  button_text: string
  /** 标题（着急版，第 3 次起） */
  title_urgent: string
  /** 正文文案（着急版） */
  message_urgent: string
  /** 按钮文案（着急版） */
  button_text_urgent: string
}

/** 关于页彩蛋文案：点击蓝色星球触发的隐藏弹窗 */
export interface EasterEggContent {
  /** 彩蛋弹窗的庆祝文案 */
  message: string
  /** 悬浮召唤按钮文案（关于页右下角”召唤星际猫咪”） */
  button_text: string
  /** 点击召唤按钮后的 Toast 提示文案 */
  toast: string
}

/** 平台元信息：作者、联系方式、GitHub 等 */
export interface PlatformContent {
  title: string
  author: string
  email: string
  github: string
}

/** AI 助手页文案 */
export interface AIAssistantContent {
  /** 输入框底部提示小字 */
  input_hint: string
}

export interface SiteContent {
  hero: HeroContent
  quick_entries: QuickEntryContent[]
  guide_steps: GuideStepContent[]
  /** 注册策略提示文案（可选：后端未返回时不阻断 HomeView 默认文案） */
  registration?: RegistrationContent
  /** 账户未激活提示文案（可选） */
  activation?: ActivationContent
  /** 关于页彩蛋文案（可选） */
  easter_egg?: EasterEggContent
  /** 平台元信息（可选） */
  platform?: PlatformContent
  /** AI 助手页文案（可选） */
  ai_assistant?: AIAssistantContent
}

File: src/omichub/application/schemas/site_content.py
"""站点首页文案 DTO（外置 YAML 驱动）"""

from omichub.application.schemas.base import OmicsHubBaseSchema


class HeroContent(OmicsHubBaseSchema):
    """Hero 欢迎区文案"""

    title: str
    description: str


class QuickEntryContent(OmicsHubBaseSchema):
    """快捷入口兜底文案。

    首页实际展示优先使用 site_settings.home_quick_entries；
    当 DB 配置不可用时，前端按 key 回退到默认路由与图标。
    """

    key: str
    title: str
    desc: str


class GuideStepContent(OmicsHubBaseSchema):
    """使用引导步骤文案"""

    title: str
    desc: str


class RegistrationContent(OmicsHubBaseSchema):
    """注册策略提示文案（公开读取）。

    真正的注册开关由后台 site-settings（DB）控制并在 /auth/register 强制校验；
    这里只提供“关闭注册时告诉用户联系谁”的文案，由 data/OmicHub.yaml 驱动。
    """

    disabled_message: str
    admin_contact: str


class ActivationContent(OmicsHubBaseSchema):
    """账户未激活提示文案（公开读取）。

    用户登录未激活账户时（auth_service 抛“账户未激活”），前端弹出友好提示，
    引导用户联系管理员激活。文案由 data/OmicHub.yaml 的 activation 段驱动，改文件即生效。
    admin_contact 缺省时回退到 registration.admin_contact，避免重复配置。

    支持“点击计数 + 文案切换”彩蛋：第 1-2 次显示正常版（title/button_text/message），
    第 3 次及以后切换为着急版（title_urgent/button_text_urgent/message_urgent），
    第 5 次点击关闭时触发全局猫爪彩蛋事件。
    """

    message: str
    admin_contact: str = ""
    title: str = "账号还在星尘中沉睡 ✨"
    button_text: str = "收到喵！"
    title_urgent: str = "喵～ 你好像很着急呢 ⏳"
    message_urgent: str = "(｡•́︿•̀｡) 再戳管理员一下嘛！开通后就能开心逛平台啦～ ✨"
    button_text_urgent: str = "我这就去！"


class EasterEggContent(OmicsHubBaseSchema):
    """关于页彩蛋文案（公开读取）。

    在「关于」页点击蓝色星球触发的隐藏弹窗，文案由 data/OmicHub.yaml 的
    easter_egg 段驱动，改文件即生效。
    """

    message: str = "🎉 恭喜你发现了 OmicHub 的隐藏星际守护者！"
    button_text: str = "🐾 召唤星际猫咪"
    toast: str = "🎉 星际守护者已响应召唤！快看看屏幕上留下的足迹吧～ ✨"


class PlatformContent(OmicsHubBaseSchema):
    """平台元信息（公开读取）。

    由 data/OmicHub.yaml 的 platform 段驱动，提供作者、联系方式、GitHub 等。
    """

    title: str = "OmicHub"
    author: str = ""
    email: str = ""
    github: str = ""


class AIAssistantContent(OmicsHubBaseSchema):
    """AI 助手页文案（公开读取）。

    由 data/OmicHub.yaml 的 ai_assistant 段驱动，改文件即生效。
    """

    input_hint: str = "内容由 AI 生成，请仔细甄别"


class SiteContentDTO(OmicsHubBaseSchema):
    """首页文案（公开读取）"""

    hero: HeroContent
    quick_entries: list[QuickEntryContent]
    guide_steps: list[GuideStepContent]
    registration: RegistrationContent
    activation: ActivationContent = ActivationContent(
        message="你的账号尚未激活，暂无法登录。请联系管理员为你开通激活，开通后即可正常使用平台～",
        admin_contact="",
    )
    easter_egg: EasterEggContent = EasterEggContent()
    platform: PlatformContent = PlatformContent()
    ai_assistant: AIAssistantContent = AIAssistantContent()
